mlir/test/Gradient/BufferizationTest.mlir

```diff
@@ -28,26 +28,45 @@ func.func @adjoint(%arg0: f64, %arg1: index) {
     return
 }
 
-///
+// -----
 
 func.func private @circuit2(%arg0: f64)
 
 func.func @backprop(%arg0: f64, %arg1: index) {
 
     // CHECK:   [[alloc:%.+]] = memref.alloc(%arg1) : memref<?xf64>
-    // CHECK:   gradient.backprop @circuit2(%arg0) size(%arg1) in([[alloc]] : memref<?xf64>) : (f64) -> ()
-    %grad = gradient.backprop @circuit2(%arg0) size(%arg1) : (f64) -> tensor<?xf64>
+    // CHECK:   gradient.backprop @circuit2({{.*}}) argnums(dense<0> : tensor<1xindex>) size(%arg1) in([[alloc]] : memref<?xf64>) : (f64) -> ()
+    %grad = gradient.backprop @circuit2(%arg0) argnums(dense<0> : tensor<1xindex>) size(%arg1) : (f64) -> tensor<?xf64>
```

Comment:
In the IR the attribute would then be passed like this (in the absence of custom assembly):
```suggestion
    %grad = gradient.backprop @circuit2(%arg0) size(%arg1) {diffArgIndices = dense<0> : tensor<1xindex>} : (f64) -> tensor<?xf64>

Reply:
Yes perfect I update that!